pkg/controllers/provisioning/provisioner.go

```diff
@@ -331,6 +332,9 @@ func (p *Provisioner) Schedule(ctx context.Context) (scheduler.Results, error) {
 			p.cluster.MarkPodSchedulingDecisions(lo.SliceToMap(pods, func(p *corev1.Pod) (*corev1.Pod, error) {
 				return p, fmt.Errorf("no nodepools found")
 			}), pods...)
+			p.cluster.DeletePodHealthyNodePoolScheduledTime(lo.SliceToMap(pods, func(p *corev1.Pod) (*corev1.Pod, error) {
+				return p, fmt.Errorf("no nodepools found")
+			}), pods...)
```

Comment:
Can we do this `lo.SliceToMap` one time?

Reply:
Refactored the code.